docs/guide/rules-concept.md

```diff
@@ -0,0 +1,181 @@
+### Validation rule handlers
```

Comment:
Need intro: what is a handler i.e. rule defines validation options. Handler executes the validation itself.